lib/handler/redirect.c

```diff
@@ -24,17 +24,23 @@
 
 #define MODULE_NAME "lib/handler/redirect.c"
 
+typedef H2O_VECTOR(char *) char_vec;
+
 struct st_h2o_redirect_handler_t {
     h2o_handler_t super;
     int internal;
     int status;
     h2o_iovec_t prefix;
+    char_vec wildcard_chars;
```

Comment:
I agree that this approach works, but I'm afraid that the design might be fragile.

The problem is that `prefix` _can_ be used directly as it has been, *unless wildcard_chars is non-empty*. I do not think people (including us) making changes to the redirect handler can make sure that that is done all the time.

Therefore, I might suggest changing `prefix` to a list (i.e. `h2o_iovec_vector_t`). When the size of the list is greater than 1, it means that `*` was within the elements, and that it should be interpolated.

WDYT?

Reply:
done in https://github.com/h2o/h2o/pull/2186/commits/508f5b3f12bb37affaba40283981a73b1565e955